core/trino-main/src/main/java/io/trino/operator/output/BlockBuilderPositionsAppender.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator.output;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.block.BlockBuilderStatus;
+import io.trino.spi.block.DictionaryBlock;
+import io.trino.spi.type.Type;
+import io.trino.type.BlockTypeOperators.BlockPositionEqual;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import org.openjdk.jol.info.ClassLayout;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class BlockBuilderPositionsAppender
```

Comment:
We don't really want to use `BlockBuilder` eventually, hence we should not expose it in name, but also in method param

Reply:
I'm gonna drop this class as it's not gonna be needed for dedicated `PositionAppender`s

Reply:
actually, there is some boilerplate code that is handled by BlockBuilderPositionsAppender so it is useful to have it until we move away from BlockBuilder.
I will drop this class as a part of https://github.com/trinodb/trino/issues/11558